src/libraries/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -238,6 +238,7 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFramework)' != '$(NetCoreAppCurrent)'">
     <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\DynamicDependencyAttribute.cs" />
+    <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\DynamicallyAccessedMembersAttribute.cs" />
```

Comment:
@eerhardt what do we expect to do here for configurations that are not net5.0? Is the plan to compile them like he is doing here or should we instead have one implementation with the attributes and one without, or should we ifdef the code?

Reply:
FWIW - having two implementations would add the maintenance cost of making sure they are always in sync. ifdef-ing is a bit better but IMO, the current approach is the cleanest/least invasive from a library author/maintainer point of view.

Reply:
> what do we expect to do here for configurations that are not net5.0?

This was discussed here: https://github.com/dotnet/runtime/issues/36656#issuecomment-630345623

For this case, I personally would just `#if` around the attributes in the existing `System.Text.Json.cs` file, just like we do here:

https://github.com/dotnet/runtime/blob/9de1a7e044b70f509c3a974ec717a94486e5391b/src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs#L671-L674

That way we aren't duplicating everything. Note there will be more annotations in System.Text.Json in the future, for example `JsonConverterAttribute` and `JsonConverter<T>`.

Reply:
I've updated the ref file to use `#ifdef`s.